### PR TITLE
Google sign in

### LINKED ISSUE #6: Implement Google Sign-in functionality
Implement Google sign-in functionality. You'd need the access to the Firebase project to implement this functionality.
Request for access on #1.
This issue is currently blocked by #3

### PR DESCRIPTION
**Pull request by Monik09**
Fixes #6 

Added Sign In functionality, Sign Out functionality and snackBar(to show user logged in).

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 org.gradle.jvmargs=-Xmx1536M
-android.enableR8=true
 android.useAndroidX=true
 android.enableJetifier=true
+android.enableR8=true
```
Comment (thecodepapaya) [inline android/gradle.properties:4]: Was this change necessary or is it accidental?

File: lib/util/google_sign_in.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    if (this.name != null) return this.name;
+    return ' ';
+  }
+
+  logout() async {
+    try {
+      await _googleSignIn.signOut();
+      // this.name = _googleSignIn.currentUser.displayName;
+      // print("Bye, $name");
+      Navigator.of(context).popAndPushNamed(LoginScreen.routeName);
```
Comment (thecodepapaya) [inline lib/util/google_sign_in.dart:42]: Just for reference, [this is another method](https://dev.to/prakashselvaraj/google-sign-in-flutter-firebase-c60) for handling sign-in using streams and listening to changes to user authentication without having to explicitly handle routes and navigators.

**Review by thecodepapaya — APPROVED** [review]

File: android/gradle.properties (hunk 1 above)
Comment (Monik09) [reply android/gradle.properties:4]: it was giving warning in android studio, so I remove that but later on, it was added again by studio.

File: lib/util/google_sign_in.dart (hunk 2 above)
Comment (Monik09) [reply lib/util/google_sign_in.dart:42]: Thank you @thecodepapaya sir for sharing this info.
